### PR TITLE
Update everything to at least v0.7

### PR DESCRIPTION
**Pull request by Crghilardi**
This PR updates the whole package so that tests pass on at least 0.7 with the new Pkg.

I used 0.7 for the deprecation warnings and did the bare minimum just to make it compatible with new Julia syntax.

I think the CI badge is broken or pointing to the wrong repo, but that's a different task.
I didn't dig into fixing the travis.yml yet to make those changes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
+#  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get update -qq; fi
+#  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install -qq x11-apps libxt6 libxrender1 libxext6 libgl1-mesa-glx qtbase5-dev qtdeclarative5-dev; fi
+#  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install -y --no-install-recommends texlive-fonts-recommended texlive-latex-extra texlive-fonts-extra dvipng texlive-latex-recommended; fi
+#  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then export GKSwstype=svg; fi
+
 script:
-  - julia -e 'Pkg.init(); Pkg.clone(pwd()); Pkg.test("OctTrees", coverage=true)'
-after_success:
-- julia -e 'cd(Pkg.dir("OctTrees")); Pkg.add("Coverage"); using Coverage; Coveralls.submit(Coveralls.process_folder())'
+  - julia --project --check-bounds=yes -e 'import Pkg; Pkg.build(); Pkg.test("Octrees"; coverage=true)'
```
Comment (sjkelly) [inline .travis.yml]: @Crghilardi Welcome and thanks for the contribution! I think this line could be removes

Comment (Crghilardi) [reply .travis.yml]: ahh thank you, missed that. Looks like travis CI is up and working again now

**Comment by sjkelly** [comment]
Awesome, thanks for the contribution! Have you tested how the performance compares to https://github.com/rdeits/RegionTrees.jl?

Give me a heads up when you want to merge.

**Comment by Crghilardi** [comment]
I snuck in one last commit attributing the original author and fixing the coverage badge. Please merge whenever possible.

I did not do any benchmarking, this was more of a revival effort. I think it's important having the Org packages reasonably up to date. 

 Not sure how it compares with `RegionTrees`? I don't know what a good benchmark would look like?

**Pull request merged**